diff --git a/src/journal/journal-file.c b/src/journal/journal-file.c
index 912eb94d0..a42a08e7a 100644
--- a/src/journal/journal-file.c
+++ b/src/journal/journal-file.c
@@ -350,6 +350,9 @@ static int journal_file_verify_header(JournalFile *f) {
                         log_debug("Journal file %s has unknown state %i.", f->path, state);
                         return -EBUSY;
                 }
+
+                if (f->header->field_hash_table_size == 0 || f->header->data_hash_table_size == 0)
+                         return -EBADMSG;
         }
 
         f->compress_xz = JOURNAL_HEADER_COMPRESSED_XZ(f->header);
